### ISSUE TITLE
Android setSimultaneousDetection not working for Scale/Rotate

### ISSUE THREAD

**Issue by nomizodiac** [tangrams/tangram-es#627]
In TouchInput class there are methods to handle SimultaneousDetection between different gestures. What i want to acheive is that If map is scaled first then it should not rotate untill scale finished. But if it rotate first then it can both scale/rotate. For that i set following flag in MapController class

`       touchInput.setSimultaneousDetectionAllowed(Gestures.SCALE, Gestures.ROTATE, false); `

After setting this flag i get the required functionality that it is scaling without rotating but then it also doesn't rotate at all.

**Comment by tallytalwar** [comment]
Ok, I am able to reproduce this behavior.

And I think whats happening is scale gesture (android in-build) detection dominates over the thresholds we have set for rotation gesture detection in the almeros gesture detection library we are using. So scale gesture is always getting detected before rotation gesture is detected.

Will have to look more into the `almeros` gesture detection library we are using for `rotation`, `shove` gestures.

**Comment by blair1618** [comment]
@nomizodiac I'm guessing that you're using a Samsung phone, yes?

Unfortunately Samsung's version(s) of Android have custom versions of the built-in Android gesture detectors, which can cause problems for handling simultaneous gestures because the Samsung gesture detectors tend to be far more sensitive to small gestures. The behavior you're trying to achieve will work just fine on a non-Samsung phone. 

Our options to deal with this are limited. We don't have the source code to Samsung's gesture detectors and they could change with any software update, so it's hard for us to reliably determine which type of gesture will be recognized first. One option is to not rely on built-in gesture detectors at all and keep all of the gesture code we need in our source tree, but this would add the burden of maintaining that code ourselves.

**Comment by nomizodiac** [comment]
@blair1618 Yes you are right i am testing on Samsung S5 but i have also tested it on Huawei Honor 4C CHM-U01 and it has the same behavior. But I can achieve this functionality on Google Maps or OpenStreet Maps at my Samsung device or other devices. What do you suggest? Can we reduce the sensitivity of scale gesture or somehow prioritize the rotate gesture detection?

**Comment by blair1618** [comment]
Still thinking about this. The options so far are:
 - Maintain a ScaleGestureDetector implementation of our own, possibly based on the AOSP version. 
 - Add additional logic in TouchInput to require some threshold of scale change before a scale gesture is officially recognized.
